commit 92db24adfa941003db1d885df01157056617f30b
Author: Maks Orlovich <maksim@kde.org>
Date:   Sun May 8 14:39:03 2011 -0400

    Fix the job-on-hold reuse logic, which caused the double-POST problem)
    
    adawit, could you please at least READ what you're backporting if you are
    going to be this aggressive? Or better yet, please don't backport anything
    that's not fixing a critical bug or is trivial, as per:
    http://techbase.kde.org/Policies/Minor_Point_Release_Policy
    
    CCMAIL: adawit@kde.org
    BUG: 272466

diff --git kio/kio/scheduler.cpp kio/kio/scheduler.cpp
index 55da053..9f5607e 100644
--- kio/kio/scheduler.cpp
+++ kio/kio/scheduler.cpp
@@ -1151,7 +1151,7 @@ Slave *SchedulerPrivate::heldSlaveForJob(SimpleJob *job)
         bool canJobReuse = (cmd == CMD_GET || cmd == CMD_MULTI_GET);
 
         if (KIO::TransferJob *tJob = qobject_cast<KIO::TransferJob *>(job)) {
-            canJobReuse = cmd == (canJobReuse || cmd == CMD_SPECIAL);
+            canJobReuse = (canJobReuse || cmd == CMD_SPECIAL);
             if (canJobReuse) {
                 KIO::MetaData outgoing = tJob->outgoingMetaData();
                 const QString resume = outgoing.value("resume");
